src/app/file-manager/file-download-modal/file-download-modal.component.ts

```diff
@@ -92,4 +209,68 @@ export class FileDownloadModalComponent  {
       summary: warnMessage,
     });
   }
+
+  // This function filters repeated folders when a parent
+  // is all-selected and there are sub-folders in the list
+  filterFolderItems(sortedItems): Item[] {
+
+    const folderItems = sortedItems.filter(item => item.indeterminate === false);
+
+    const singleItems = [];
+
+    for (let i = 0; i < folderItems.length;) {
+      const parent = folderItems[i];
+
+      const isDad = this.isParent(parent, folderItems[i + 1]);
+
+      if (!singleItems.includes(parent)) {
+        singleItems.push(parent);
+      }
+
+      if (isDad) {
+        folderItems.splice(i + 1, 1);
+      } else {
+        i++;
+      }
+    }
+    return singleItems;
+  }
+
+  isParent(parent: Item, child: Item): boolean {
+
+    if (parent.type === 'File') {
+      return false;
+    }
+
+    if (child === undefined) {
+      return false;
+    }
+
+    return child.path.startsWith(parent.path);
+  }
+
+  cloneItem(item: Item): Item {
```

Comment:
It looks like this should be a method on `Item`

Reply:
Right now is a file-download-modal class specific method, makes sense only there, no need to be included in Item class I think.